src/Shared/CopyOnWriteDictionary.cs

```diff
@@ -227,6 +227,18 @@ public void Add(string key, V value)
             _backing = _backing.SetItem(key, value);
         }
 
+        /// <summary>
+        /// Adds several value to the dictionary.
+        /// </summary>
+        public void SetItems(IEnumerable<KeyValuePair<string, V>> items)
+        {
+            _backing = _backing.SetItems(items);
```

Comment:
`SetItems` and `Where` will also need to be implemented on the .NET 3.5 version of `ImmutableDictionary` in C:\src\msbuild\src\MSBuildTaskHost\Immutable\ImmutableDictionary.cs.

Reply:
Good catch! Why isn't `Where` implemented by LINQ as it is for most IEnumerables?

Reply:
Of course! I stand corrected, only `SetItems` is missing.